app/code/Magento/Catalog/Model/Product/Image/Cache.php

```diff
@@ -127,4 +154,97 @@ protected function processImageData(Product $product, array $imageData, $file)
 
         return $this;
     }
+
+    /**
+     * Get themes in use
+     *
+     * @return array
+     */
+    private function getThemesInUse(): array
```

Comment:
Consider refactoring this logic with introducing composite object (with configuration via DI) for themes in use and move logic to corresponded modules (pages to CMS module; product, category -> Catalog module etc)